pkg/apis/controlplane/v1beta2/types.go

```diff
@@ -0,0 +1,292 @@
+	IP IPAddress `json:"ip,omitempty" protobuf:"bytes,2,opt,name=ip"`
+	// Ports maintain the named port mapping of this Pod.
+	Ports []NamedPort `json:"ports,omitempty" protobuf:"bytes,3,rep,name=ports"`
+}
+
+// ExternalEntityReference represents a ExternalEntity Reference.
+type ExternalEntityReference struct {
+	// The name of this ExternalEntity.
+	Name string `json:"name,omitempty" protobuf:"bytes,1,opt,name=name"`
+	// The namespace of this ExternalEntity.
+	Namespace string `json:"namespace,omitempty" protobuf:"bytes,2,opt,name=namespace"`
+}
+
+// GroupMember represents resource member to be populated in Groups.
+// This supersedes GroupMemberPod, and will eventually replace it.
+type GroupMember struct {
+	// Pod maintains the reference to the Pod.
+	Pod *PodReference `json:"pod,omitempty" protobuf:"bytes,1,opt,name=pod"`
+	// ExternalEntity maintains the reference to the ExternalEntity.
+	ExternalEntity *ExternalEntityReference `json:"externalEntity,omitempty" protobuf:"bytes,2,opt,name=externalEntity"`
+	// IP is the IP address of the Endpoints associated with the GroupMember.
+	IPs []IPAddress `json:"ips,omitempty" protobuf:"bytes,3,rep,name=ips"`
+	// Ports is the list NamedPort of the GroupMember.
+	Ports []NamedPort `json:"ports,omitempty" protobuf:"bytes,4,rep,name=ports"`
+}
+
+// +k8s:deepcopy-gen:interfaces=k8s.io/apimachinery/pkg/runtime.Object
+// AppliedToGroupPatch describes the incremental update of an AppliedToGroup.
+type AppliedToGroupPatch struct {
+	metav1.TypeMeta     `json:",inline"`
+	metav1.ObjectMeta   `json:"metadata,omitempty" protobuf:"bytes,1,opt,name=metadata"`
+	AddedPods           []GroupMemberPod `json:"addedPods,omitempty" protobuf:"bytes,2,rep,name=addedPods"`
+	RemovedPods         []GroupMemberPod `json:"removedPods,omitempty" protobuf:"bytes,3,rep,name=removedPods"`
+	AddedGroupMembers   []GroupMember    `json:"addedGroupMembers,omitempty" protobuf:"bytes,4,rep,name=addedGroupMembers"`
+	RemovedGroupMembers []GroupMember    `json:"removedGroupMembers,omitempty" protobuf:"bytes,5,rep,name=removedGroupMembers"`
+}
+
+// +k8s:deepcopy-gen:interfaces=k8s.io/apimachinery/pkg/runtime.Object
+// AppliedToGroupList is a list of AppliedToGroup objects.
+type AppliedToGroupList struct {
+	metav1.TypeMeta `json:",inline"`
+	metav1.ListMeta `json:"metadata,omitempty" protobuf:"bytes,1,opt,name=metadata"`
+	Items           []AppliedToGroup `json:"items" protobuf:"bytes,2,rep,name=items"`
+}
+
+// +genclient
+// +genclient:nonNamespaced
+// +genclient:onlyVerbs=list,get,watch
+// +k8s:deepcopy-gen:interfaces=k8s.io/apimachinery/pkg/runtime.Object
+// AddressGroup is the message format of antrea/pkg/controller/types.AddressGroup in an API response.
+type AddressGroup struct {
+	metav1.TypeMeta   `json:",inline"`
+	metav1.ObjectMeta `json:"metadata,omitempty" protobuf:"bytes,1,opt,name=metadata"`
+	Pods              []GroupMemberPod `json:"pods,omitempty" protobuf:"bytes,2,rep,name=pods"`
```

Comment:
Do we still need Pods in v1beta2? do we plan to use GroupMembers to replace it? Is it the chance to make the change when upgrading to v1beta2?

Reply:
That could be done, but we need to do this in a separate PR since it will be huge as well

Reply:
What about Pods in appliedTo groups? Should we deprecate GroupMemberPod altogether?